docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,99 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies changes to the auth and bank modules to implement vested accounts for the Cosmos Hub. 
+The requirements for this vested account is that it should be capable of being initialized during genesis with
+a starting balance X coins and a vesting blocktime T. The owner of this account should be able to delegate to validators and vote,
+however they cannot send their coins to other accounts until the account has fully vested. Thus, the bank module's `MsgSend` handler 
+should error if a vested account is trying to send an amount before time T.
+
+### Implementation
+
+##### Changes to x/auth Module
+
+The `Account` interface will specify both the Account type and any parameters it needs.
+
+```go
+// Account is a standard account using a sequence number for replay protection
+// and a pubkey for authentication.
+type Account interface {
+    Type() string // returns the type of the account
```

Comment:
(btw, should be Vesting not Vested).

I'm not sure that having a top-level type is the right solution.
Would it make more sense to have an optional interface method, like 

```
type VestingAccount interface {
    Account
    AssertIsVestingAccount() // existence implies that account is vesting.
}
```

Then you an check with `vacc, ok := acc.(VestingAccount); ok`

Reply:
That works well in our case where we are choosing from two possible account types.

I think we should leave the option open to adding more account types in the future, and for developers building on the SDK to have many supported account types.

In those cases, I think having a `Type()` function that handlers can use to switch their functionality based on the account type is better than trying to type cast for multiple account types.